### PR TITLE
gh-94635: Normalise sqlite3 doc headings

### LINKED ISSUE #94635: Restructure the sqlite3 docs table-of-contents
The current docs are laid out more or less like this:

- short introduction
- relatively short tutorial
- reference
- how-to's/guides and some in-depth explanations

Suggesting to make the following improvements:

- add top-level headings for:
    - introduction
    - tutorial
    - reference
    - guides
    - concepts
- reorder sections according to the new ToC
- add anchors for each top-level heading

Implemented by (all backported to 3.11 and 3.10):
- #95269 
- #94636 
- #95751
- #96089
- #94677 (closed, as it was implemented by other PRs)
- #96136

### PR DESCRIPTION
**Pull request by erlend-aasland**
Fixes #94635

### CODE REVIEW

File: Doc/library/sqlite3.rst [hunk 1]
```diff
@@ -1450,8 +1450,8 @@ This section shows recipes for common adapters and converters.
 
 .. _sqlite3-controlling-transactions:
```
Comment (erlend-aasland) [inline Doc/library/sqlite3.rst:1458]: I'm not sure if we should change the anchor as well.

Comment (CAM-Gerlach) [reply Doc/library/sqlite3.rst:1458]: You could add an additional ref target with the new name, but I would suggest just leaving the old one. The whole idea behind ref targets is that they stay the same even when headings change so you don't need to worry about breaking links.

**Review by CAM-Gerlach — COMMENTED** [review]
* [x] [Per the devguide](https://devguide.python.org/documenting/#capitalization),
  
  > In the Python documentation, the use of sentence case in section titles is preferable, but consistency within a unit is more important than following this rule. If you add a section to a chapter where most sections are in title case, you can either convert all titles to sentence case or use the dominant style in the new section title.

  Since around half or more of the tiles are already sentence case, if we're going to change it, it makes sense to do so to the preferred style.

* [ ] I'm not sure adding the "How to" prefix is necessary on titles that already are framed as tasks, since it is somewhat redundant and repetitive, especially if/when the supersection title is changed to "How-to"—it is inherent than sections directly underneath a "How to" section describe "How to" do something.

* [x] While you're at it and touching these anyway, you _could_ normalize it to use the [standard heading levels in the devguide](https://devguide.python.org/documenting/#sections)—basically, just moving everything up a level, IIUC. But that's strictly optional, of course.

Other than that, this seems okay, assuming the plan is to move "Default Adapters and Converters" and "Adapter and Converter Recipes", etc., under their appropriate "How-to" sections ("Adapt..." and "Convert...") in the followup PR.

Thanks!

**Comment by erlend-aasland** [comment]
> While you're at it and touching these anyway, you could normalize it to use the [standard heading levels in the devguide](https://devguide.python.org/documenting/#sections)—basically, just moving everything up a level, IIUC. But that's strictly optional, of course.

I'm planning to address that in a separate PR. Let's do the headings first; I want these doc PRs to be concise.

**Comment by erlend-aasland** [comment]
> I'm not sure adding the "How to" prefix is necessary on titles that already are framed as tasks, since it is somewhat redundant and repetitive, especially if/when the supersection title is changed to "How-to"—it is inherent than sections directly underneath a "How to" section describe "How to" do something.

Absolutely, but Diátaxis explicitly recommends such headings: https://diataxis.fr/how-to-guides/#pay-attention-to-naming

**Review by ezio-melotti — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Regarding the "How to" prefix; perhaps we should put this PR on hold until we're through the Diátaxis workshop in August. There is no hurry to get this done anyways.

**Comment by CAM-Gerlach** [comment]
> Absolutely, but Diátaxis explicitly recommends such headings: [diataxis.fr/how-to-guides/#pay-attention-to-naming](https://diataxis.fr/how-to-guides/#pay-attention-to-naming)

Right, but that's for top-level page _titles_, and the section headings would still be worded in task-oriented "how to" form (e.g. "(How to) adapt Python types to custom SQLite values", rather than the proscribed "Adapting Python types to custom SQLite values" or "Python-SQLite type conversion", with the "How to" wording as the top-level parent section, just not also repeated again in every sub-section title underneath it.

**Comment by erlend-aasland** [comment]
I guess you're right; that makes sense. I'd still like to put this on hold until after the workshop.

**Comment by erlend-aasland** [comment]
Here's how Django has organised their How To chapter:

https://docs.djangoproject.com/en/4.0/howto/

**Comment by erlend-aasland** [comment]
Closing this, as most of the work has been done in other PR's. If we need to tweak headings, we can do so on a case-by-case basis.

**Comment by CAM-Gerlach** [comment]
There are still some parts the need doing, especially per Danielle's guidance today:

* The "Using", etc. names should be converted as here, and also the "Adapter and converter recipes" needs to be renamed to...something how-to-ish
* The "Default adapters and converters" is actually a Reference section

I can open PRs—and if needed, separate issues—for each of those

**Pull request closed without merging**